crates/oxc_linter/src/rules/eslint/no_lonely_if.rs

```diff
@@ -0,0 +1,168 @@
+    /// an `else if` instead.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// if (condition) {
+    ///   // ...
+    /// } else {
+    ///   if (anotherCondition) {
+    ///     // ...
+    ///   }
+    /// }
+    /// ```
+    ///
+    /// ```js
+    /// if (condition) {
+    ///   // ...
+    /// } else {
+    ///   if (anotherCondition) {
+    ///     // ...
+    ///   } else {
+    ///       // ...
+    ///   }
+    /// }
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// if (condition) {
+    ///   // ...
+    /// } else if (anotherCondition) {
+    ///   // ...
+    /// }
+    /// ```
+    ///
+    /// ```js
+    /// if (condition) {
+    ///   // ...
+    /// } else if (anotherCondition) {
+    ///   // ...
+    /// } else {
+    ///   // ...
+    /// }
+    /// ```
+    ///
+    /// ```js
+    /// if (condition) {
+    ///   // ...
+    /// } else {
+    ///   if (anotherCondition) {
+    ///     // ...
+    ///   }
+    ///   doSomething();
+    /// }
+    /// ```
+    NoLonelyIf,
+    eslint,
+    pedantic,
+    pending
+);
+
+impl Rule for NoLonelyIf {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::IfStatement(if_stmt) = node.kind() else {
+            return;
+        };
+
+        let Some(ref alternate) = if_stmt.alternate else {
+            return;
+        };
+
+        let BlockStatement(b) = alternate else {
+            return;
+        };
+
+        if let Some(AstKind::IfStatement(_)) = ctx.nodes().parent_node(node.id()).map(AstNode::kind)
+        {
+            return;
+        };
+
+        if b.body.len() == 1 {
+            let Some(only_stmt) = b.body.first() else {
+                return;
+            };
+
+            if let Statement::BlockStatement(inner_block) = only_stmt {
+                if inner_block.body.len() == 1 {
+                    if let Some(Statement::IfStatement(lonely_if)) = inner_block.body.first() {
+                        ctx.diagnostic(no_lonely_if_diagnostic(Span::new(
+                            lonely_if.span.start,
+                            lonely_if.span.start + 2,
+                        )));
+                    };
+                }
+            } else if let Statement::IfStatement(lonely_if) = only_stmt {
+                ctx.diagnostic(no_lonely_if_diagnostic(Span::new(
+                    lonely_if.span.start,
+                    lonely_if.span.start + 2,
+                )));
+            };
+        }
+    }
+}
+
+#[test]
+fn test() {
+    use crate::tester::Tester;
+
+    let pass = vec![
+        "if (a) {;} else if (b) {;}",
+        "if (a) {;} else { if (b) {;} ; }",
+        "if (a) if (a) {} else { if (b) {} } else {}",
+        "if (a) {
+           if (b) {} else { }
+         } else {
+           if (c) {  } 
+           if (d) {  } 
+          }",
+    ];
+
+    let fail = vec![
```

Comment:
It looks like we are missing some test cases from eslint. could you port those as well? e.g.:

```
if (foo) {
} else {
  if (bar) baz()
}
[1, 2, 3].forEach(foo);
```

Reply:
Sure thing, added them!